pages/preimages/[hash].tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Skeleton } from 'antd';
+import { GetServerSideProps } from 'next';
+import dynamic from 'next/dynamic';
+import { IPreimageData, getLatestPreimage } from 'pages/api/v1/preimages/latest';
+import React, { FC, useEffect } from 'react';
+
+import { getNetworkFromReqHeaders } from '~src/api-utils';
+import { useNetworkContext } from '~src/context';
+import SEOHead from '~src/global/SEOHead';
+import { ErrorState } from '~src/ui-components/UIStates';
+
+const PreImagesTable = dynamic(() => import('~src/components/PreImagesTable'), {
+	loading: () => <Skeleton active /> ,
+	ssr: false
+});
+
+export const getServerSideProps: GetServerSideProps = async ({ req, query }) => {
+	const { hash } = query;
+	const network = getNetworkFromReqHeaders(req.headers);
+
+	const { data, error } = await getLatestPreimage({ hash: String(hash), network });
```

Comment:
if hash is not passed in query.
This will send `'undefined'` as a string to the `getLatestPreimage()` function